dashboard/app/models/script_level.rb

```diff
@@ -67,6 +67,36 @@ def oldest_active_level
     end
   end
 
+  def has_another_level_to_go_to?
+    if script.professional_learning_course?
+      !end_of_stage?
+    else
+      next_progression_level
+    end
+  end
+
+  def next_level_or_redirect_path_for_user user
+    level_to_follow = (level.unplugged? || stage.try(:unplugged?)) ? next_level : next_progression_level
```

Comment:
We [already](https://github.com/code-dot-org/code-dot-org/blob/a0f8a46be76648b9c360b8915e308ccf5a6d34cd/dashboard/app/controllers/api_controller.rb#L151) [make](https://github.com/code-dot-org/code-dot-org/blob/1b07778364ec31b628b24853b91e514f428ede39/dashboard/app/helpers/levels_helper.rb#L140) [assumptions](https://github.com/code-dot-org/code-dot-org/blob/921ef18b524d67383fa6af150d5caf690195f308/dashboard/app/views/layouts/_header.html.haml#L70) that `stage` is non-nil elsewhere.